common/lib/xmodule/xmodule/word_cloud_module.py

```diff
@@ -237,23 +234,38 @@ def handle_ajax(self, dispatch, data):
                 'error': 'Unknown Command!'
             })
 
-    def get_html(self):
-        """Template rendering."""
-        context = {
+    def student_view(self, context):
+        """
+        Renders the output that a student will see.
+        """
+        fragment = Fragment()
+
+        fragment.add_content(self.system.render_template('word_cloud.html', {
             'ajax_url': self.system.ajax_url,
             'display_name': self.display_name,
             'instructions': self.instructions,
             'element_class': self.location.category,
             'element_id': self.location.html_id(),
             'num_inputs': self.num_inputs,
             'submitted': self.submitted,
-        }
-        self.content = self.system.render_template('word_cloud.html', context)
-        return self.content
+        }))
+
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/d3.min.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/d3.layout.cloud.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/word_cloud.js'))
+        fragment.add_javascript_url(self.runtime.local_resource_url(self, 'public/js/word_cloud_main.js'))
+
+        return fragment
+
+    def author_view(self, context):
```

Comment:
I forget if this is necessary. Isn't the default to use the student view if an author_view isn't provided?